ariadne/cardano/src/Ariadne/Wallet/Backend/KeyStorage.hs

```diff
@@ -452,6 +472,26 @@ renameSelection pwl WalletFace{..} walletSelRef name = do
 
   walletRefreshState
 
+-- | This function removes unknown keys from key file on startup.
+checkUnknownKeys
+    :: PassiveWalletLayer IO
+    -> FilePath
+    -> (ConfirmationType -> IO Bool)  -- ^ What to do with unknown keys in keyfile.
+    -> IO ()
+checkUnknownKeys pwl path confirmation = do
+    keyFile <- BS.readFile path
+    roots   <- pwlGetWalletIds pwl
+    us      <- case decodeFull' @UserSecret keyFile of
```

Comment:
We want to hide `UserSecret` behind `Keystore`. And hide its operations behind `PassiveWalletLayer`.
I propose to add a function to `PassiveWalletLayer` which checks whether the set of keys in `Keystore` is the same as the set of wallets in acid-state DB. It shouldn't use `UserSecret` directly, it should use `toList` from `Keystore`. It can return `Set HdRootId`, for example. I. e. set of `HdRootId`s which are present in one set and absent in the other one.